### DISCUSSION TITLE
Is there a right way to send images to controlnet?

### DISCUSSION THREAD

**Discussion by missionfloyd** [Mikubill/sd-webui-controlnet#1599] [Q&A]
Every extension that sends images to controlnet seems to have its own roundabout way of doing it using javascript, but is there "proper" way to do it, like webui's [register_paste_params_button()](https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/baf6946e06249c5af9851c60171692c44ef633e0/modules/generation_parameters_copypaste.py#L110-L111C17)?

**Comment by huchenlei** [comment]
You can override ControlNet unit's `image` field in `StableDiffusionProcess` to pass image to controlnet. See how ControlNet picks input image here:
https://github.com/Mikubill/sd-webui-controlnet/blob/5fc952793aa97bdd0078574b3be8bc98e1fcf2cd/scripts/controlnet.py#L561

However, this will not replace the input image on the front-end and cause confusions. So if you want to do it in JavaScript, you still need to locate the image input element of ControlNet and set the value of img src.

There is currently nothing like webui's register_paste_params_button in ControlNet extension.
